dtfs-central-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js

```diff
@@ -1,14 +1,22 @@
-const { getUtilisationReportDetails } = require('../../../services/repositories/utilisation-reports-repo');
+const { getUtilisationReportDetails, getUtilisationReportDetailsForMonthAndYear } = require('../../../services/repositories/utilisation-reports-repo');
 
 const getUtilisationReports = async (req, res) => {
   try {
     const { bankId } = req.params;
 
-    const utilisationReports = await getUtilisationReportDetails(bankId);
+    const { month, year } = req.query;
+
+    let utilisationReports;
+    if (month && year) {
+      const utilisationReport = await getUtilisationReportDetailsForMonthAndYear(bankId, month, year);
+      utilisationReports = utilisationReport ? [utilisationReport] : [];
+    } else {
+      utilisationReports = await getUtilisationReportDetails(bankId);
+    }
 
     res.status(200).send(utilisationReports);
   } catch (error) {
-    console.error('Unable to get utilisation reports %s', error);
+    console.error('Unable to get utilisation reports: ', error);
```

Comment:
Nit: can remove the space in `reports: `. When you just add `error` as a second argument this is done for you.